frontend/awx/access/credentials/hooks/useCredentialsFilters.tsx

```diff
@@ -1,25 +1,12 @@
-import { useMemo } from 'react';
-import { IToolbarFilter } from '../../../../../framework';
-import {
-  useCreatedByToolbarFilter,
-  useDescriptionToolbarFilter,
-  useModifiedByToolbarFilter,
-  useNameToolbarFilter,
-} from '../../../common/awx-toolbar-filters';
+import { useDynamicToolbarFilters } from '../../common/useDynamicFilters';
+import { Credential } from '../../../interfaces/Credential';
 
 export function useCredentialsFilters() {
-  const nameToolbarFilter = useNameToolbarFilter();
-  const descriptionToolbarFilter = useDescriptionToolbarFilter();
-  const createdByToolbarFilter = useCreatedByToolbarFilter();
-  const modifiedByToolbarFilter = useModifiedByToolbarFilter();
-  const toolbarFilters = useMemo<IToolbarFilter[]>(
-    () => [
-      nameToolbarFilter,
-      descriptionToolbarFilter,
-      createdByToolbarFilter,
-      modifiedByToolbarFilter,
-    ],
-    [nameToolbarFilter, descriptionToolbarFilter, createdByToolbarFilter, modifiedByToolbarFilter]
-  );
-  return toolbarFilters;
+  const toolBarFilters = useDynamicToolbarFilters<Credential>({
+    optionsPath: 'credentials',
+    preSortedKeys: ['name'],
+    preFilledValueKeys: ['name', 'credential_type'],
```

Comment:
Let's add `credential_type` to `knownAwxFilterKeys` like so:

```
credential_type: {
    resourceType: 'credential_type',
    params: { order_by: '-created' },
    labelKey: 'name',
    valueKey: 'id',
  }, 
```

That way the dropdown values will be a list of credential type names instead of their IDs.